### PR TITLE
#RPR-52 updated flow, fixed bugs and added the ad refresher plugin

### PR DESCRIPTION
**Pull request by linorabolini**
Hello ! 

Quick update to get the ads refreshed. I made an update to the google tag related events so they emit the element + a data prop as well. In this way it is more generic and its to prepare the ground to abstract the googletag ad server module.

The rest is just logic migration, there is many logic that seems only related to custom logging and WP targeting update. I'm thinking that for those cases we may want to emit new events like NODE_REFRESH (before and after refresh), so we can allow acting upon these events from the plugins instead of a callback in a plugin. 

Thoughts?

### CODE REVIEW

File: src/core/plugins/adRefresher/index.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export default (delay, cb, cap) => core => {
+    core.on(core.SLOT_RENDER_ENDED, function g(e, { element }) {
```
Comment (eric-washpost) [inline src/core/plugins/adRefresher/index.js]: refresh only happens when the impression viewed. Please check the refresh function on the original Zeus. We did some updates base on the creative sizes.

File: src/index.js [hunk 2]
```diff
@@ -29,6 +30,24 @@ const { inventory = {}, sraBundles = {} } = window.wpAd || {}
 const groups = Object.keys(sraBundles).map(key => sraBundles[key])
 const { commercialNode } = window
 
+// washpost ad refresh config
+// TODO: move this to a WP plugin ?
+// TODO: maybe trigger an event when slot is refreshed: NODE_BEFORE_REFRESH
+const wpAdRefresher = dt => (
```
Comment (eric-washpost) [inline src/index.js]: could you rename it? we don't want to have wpAd keywords on the functions.

File: src/core/plugins/adRefresher/index.js (hunk 1 above)
Comment (linorabolini) [reply src/core/plugins/adRefresher/index.js]: ahhh great catch ! will update it !

File: src/index.js (hunk 2 above)
Comment (linorabolini) [reply src/index.js]: its renamed in RPR-55 when I restructured the folders based on the feedback of our talk :) ! 

I'm also adding url params to deactivate refresh based on our today's talk !

**Pull request merged**